core/test/network-records-to-devtools-log.js

```diff
@@ -227,6 +244,48 @@ function addRedirectResponseIfNeeded(networkRecords, record) {
 function networkRecordsToDevtoolsLog(networkRecords, options = {}) {
   const devtoolsLog = [];
   networkRecords.forEach((networkRecord, index) => {
+    // If we're operating on network record raw objects (not NetworkRequest instances),
+    // then we're operating on test data that may need to be massaged a bit.
+    if (networkRecord.constructor === Object) {
+      // Temporary code while we transition away from startTime and endTime.
+      // This allows us to defer slightly changes to test files.
+      // TODO: remove after timing refactor is done
+      // See https://github.com/GoogleChrome/lighthouse/pull/14311
+      if (networkRecord.startTime !== undefined) {
+        networkRecord.networkRequestTime = networkRecord.startTime;
+        // Old tests never distinguished between these two.
+        networkRecord.rendererStartTime = networkRecord.startTime;
+      }
+      if (networkRecord.endTime !== undefined) {
+        networkRecord.networkEndTime = networkRecord.endTime;
+      }
+      if (networkRecord.responseReceivedTime !== undefined) {
+        networkRecord.responseHeadersEndTime = networkRecord.responseReceivedTime;
+      }
+
+      // Set timing.requestTime and timing.receiveHeadersEnd to be values that
```

Comment:
I thought I could add a precondition here:

```js
if (record.rendererStartTime === undefined ^ record.networkEndTime === undefined) {
        // eslint-disable-next-line max-len
        throw new Error('expected test record to define both rendererStartTime and networkEndTime, but never just one');
      }
```

But 10 tests threw errors because they mocked network records and only defined startTime (or endTime). Should I update those tests so we can ensure to more realistic values here?

Reply:
actually all are from uses-rel-preload, and all just set startTime leaving endTime undefined.

it'd be easy to set endTime to something in those tests (-1, I guess that is the implicit default getting used?)

or maybe assign endTime -1 if startTime is defined and endTime is missing on a test record object?

Reply:
it's fine to update the tests, but we do want some solution in here for future test writers, either picking a reasonable value or cause some failure (e.g. in the roundtripping) so the test author knows something is wrong

Reply:
I thought this would work:

```
if (record.rendererStartTime === undefined) {
        record.rendererStartTime = 0;
      }
      if (record.networkEndTime === undefined) {
        record.networkEndTime = -1;
      }
```

But had complications getting it to work w/ the `redirectResponseTimestamp` kludge used elsewhere. I'm getting a bit off track here, so gonna forget about normalizing the test data for now.